LOGIC_GATES/LOOPS/LOOP/LOOP.py

```diff
@@ -70,12 +70,23 @@ def LOOP(
 
     node_id = default_params.node_id
 
+    loop_data: LoopData = load_loop_data(node_id, num_loops)
+
+    # given the addition of the break node, it is possible that
+    # another node can write to the data of this loop. we have to
+    # now check if that's the case, and if so, return
+    if loop_data.get_data().get("is_finished"):
+        return build_result([default] if default else [], True)
```

Comment:
Could this potentially break nested loops? This check will always be executed before the `.restart()` so the loop will just exit immediately upon being reached again (?)

Reply:
I demoed it on nested loops and it worked well. Soon as loop starts, it writes its data for all nodes to see, and this just catches if other nodes have written to its data in `SmallMemory`. Each loop node has its unique data that the `break` node can see, so you're able to uniquely change a particular instance of a `loop` node and thus break only that loop.

Reply:
I see, the inner loop is able to restart properly on subsequent iterations of the outer loop?

Reply:
It is, but what I can do to be extra safe is just add a line to explicitly write `is_finished=False` to the memory of that node to ensure clean restart, probably a good idea